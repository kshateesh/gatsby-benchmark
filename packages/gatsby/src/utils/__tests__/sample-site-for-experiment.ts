import sampleSiteForExperiment from "../sample-site-for-experiment"

describe(`sampleSiteForExperiment`, () => {
  it(`returns true or false depending on if they randomly are bucketed in or not`, () => {
    const experiments = [`a`, `b`, `c`, `d`, `e`, `f`, `h`, `i`]
    experiments.forEach(experiment => {
      expect(sampleSiteForExperiment(experiment, 0)).toBeFalsy()
      expect(sampleSiteForExperiment(experiment, 50)).toMatchSnapshot()
      expect(sampleSiteForExperiment(experiment, 100)).toBeTruthy()
    })
  })
})
