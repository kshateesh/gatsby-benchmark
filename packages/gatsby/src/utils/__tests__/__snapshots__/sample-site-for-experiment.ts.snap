// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sampleSiteForExperiment returns true or false depending on if they randomly are bucketed in or not 1`] = `true`;

exports[`sampleSiteForExperiment returns true or false depending on if they randomly are bucketed in or not 2`] = `false`;

exports[`sampleSiteForExperiment returns true or false depending on if they randomly are bucketed in or not 3`] = `true`;

exports[`sampleSiteForExperiment returns true or false depending on if they randomly are bucketed in or not 4`] = `true`;

exports[`sampleSiteForExperiment returns true or false depending on if they randomly are bucketed in or not 5`] = `false`;

exports[`sampleSiteForExperiment returns true or false depending on if they randomly are bucketed in or not 6`] = `true`;

exports[`sampleSiteForExperiment returns true or false depending on if they randomly are bucketed in or not 7`] = `false`;

exports[`sampleSiteForExperiment returns true or false depending on if they randomly are bucketed in or not 8`] = `false`;
